### ISSUE TITLE
ice_quiver.py and ice_quiver_i.sh problems in the IsoSeq pipeline

### ISSUE THREAD

**Issue by contig** [PacificBiosciences/SMRT-Analysis#162]
Hello,

I believe there is a problem in the ice_quiver_i.sh script generated by the IsoSeq pipeline (version 2.3.0, run from command line), which is similar to the problem in ice_partial_i.sh (as described in Issue # 141). 

Specifically, the problem is in the call to ice_quiver.py:

```
# Task 3
ice_quiver.py -v i /home/olga/small_iso_run_16Dec2014/data/clusterOutDir $i 14 --bas_fofn /home/olga/small_iso_run_16Dec2014/input.fofn --fasta_fofn /home/olga/small_iso_run_16Dec2014/input.fasta.fofn --quiver_nproc 8 --blasr_nproc 2 || exit $?
```

Above, the variable $i contains several values ( in my case 0 through 13), but the call is expecting only one value. There needs to be a loop at some point that would go through all the values of the index i. 

I've tried running the above call to ice_quiver.py i manually, but there also seems to be a problem with how the input arguments to this function are parsed. I've tried looking at the parser definitions in ToolRunner.py, but did not get very far. 

For reference, here is the full ice_quiver_i.sh script that is generated when I run the IsoSeq smrtpipe.py pipeline:

```
#!/bin/bash
# Setting up SMRTpipe environment
echo "Setting up ENV on $(uname -n)" for task ice_quiver_i

SEYMOUR_HOME=/home/Shared/PacBio/smrt_2.3.0/install/smrtanalysis_2.3.0.140936
source $SEYMOUR_HOME/etc/setup.sh

# Create the local TMP dir if it doesn't exist
tmp_dir=$(readlink -m "/home/olga/scratch")
if [ ! -e "$tmp_dir" ]; then
   stat=0
   mkdir -p $tmp_dir || stat=$?
   if [[ $stat -ne 0 ]]; then
       echo "SMRTpipe Unable to create TMP dir '/home/olga/scratch' on $(uname -n)" 1>&2
       exit 1
   else
       echo "successfully created or found TMP dir '/home/olga/scratch'"
   fi
elif [[ ! -d "$tmp_dir" ]]; then
   echo "SMRTpipe TMP /home/olga/scratch must be a directory on $(uname -n)" 1>&2
   exit 1
fi

########### TASK metadata #############
# Task            : ice_quiver_i
# Module          : P_IsoSeqCluster
# Module Version  : 2.3.140629
# TaskType        : None
# URL             : task://Anonymous/P_IsoSeqCluster/ice_quiver_i
# createdAt       : 2014-12-16 13:29:03.461229
# createdAt (UTC) : 2014-12-16 12:29:03.461260
# ncmds           : 4
# LogPath         : /home/olga/small_iso_run_16Dec2014/log/P_IsoSeqCluster/ice_quiver_i.log
# Script Path     : /home/olga/small_iso_run_16Dec2014/workflow/P_IsoSeqCluster/ice_quiver_i.sh

# Input       : /home/olga/small_iso_run_16Dec2014/input.fasta.fofn
# Input       : /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_partial_merge.sentinel
# Input       : /home/olga/small_iso_run_16Dec2014/input.fofn
# Output      : /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_quiver_i.sentinel
#
########### END TASK metadata #############

cd /home/olga/small_iso_run_16Dec2014/log/P_IsoSeqCluster
# Writing to log file
cat /home/olga/small_iso_run_16Dec2014/workflow/P_IsoSeqCluster/ice_quiver_i.sh >> /home/olga/small_iso_run_16Dec2014/log/P_IsoSeqCluster/ice_quiver_i.log;



echo "Running task://Anonymous/P_IsoSeqCluster/ice_quiver_i on $(uname -a)"

echo "Started on $(date -u)"
echo 'Validating existence of Input Files'
if [ -e /home/olga/small_iso_run_16Dec2014/input.fasta.fofn ]
then
echo 'Successfully found /home/olga/small_iso_run_16Dec2014/input.fasta.fofn'
else
echo 'WARNING: Unable to find necessary input file, or dir /home/olga/small_iso_run_16Dec2014/input.fasta.fofn.'
fi
if [ -e /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_partial_merge.sentinel ]
then
echo 'Successfully found /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_partial_merge.sentinel'
else
echo 'WARNING: Unable to find necessary input file, or dir /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_partial_merge.sentinel.'
fi
if [ -e /home/olga/small_iso_run_16Dec2014/input.fofn ]
then
echo 'Successfully found /home/olga/small_iso_run_16Dec2014/input.fofn'
else
echo 'WARNING: Unable to find necessary input file, or dir /home/olga/small_iso_run_16Dec2014/input.fofn.'
fi
echo 'Successfully validated input files'

# Task ice_quiver_i commands:


# Completed writing Task ice_quiver_i commands


# Task 1
i=`cat /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_partial_merge.sentinel` || exit $?
echo "Task 1 completed at $(date)"
# Task 2
echo i=$i || exit $?
echo "Task 2 completed at $(date)"
# Task 3
ice_quiver.py -v i /home/olga/small_iso_run_16Dec2014/data/clusterOutDir $i 14 --bas_fofn /home/olga/small_iso_run_16Dec2014/input.fofn --fasta_fofn /home/olga/small_iso_run_16Dec2014/input.fasta.fofn --quiver_nproc 8 --blasr_nproc 2 || exit $?
echo "Task 3 completed at $(date)"
# Task 4
touch /home/olga/small_iso_run_16Dec2014/data/sentinel/ice_quiver_i.sentinel || exit $?
echo "Task 4 completed at $(date)"



rcode=$?
echo "Finished on $(date -u)"
echo "Task ice_quiver_i with nproc 8 with exit code ${rcode}."
exit ${rcode}
```

**Comment by pb-jlandolin** [comment]
Hi Contig,

We just fixed this problem (Internal bug #25943), but the release will not be out for a month or so.  

To work around it, you need to run IsoSeq in distributed mode (per our minimum requirements), or you can re-run the job once the next patch is released.

The problem has to do with how P_IsoSeqCluster.py handles execution on a single node:

in ice_partial_i.log

The output from Task 1:
`cat ice_partial_split.sentinel` is:

```
0 1 2 3 4 5

which is the input for 
Task 3:
ice_partial.py -v i clusterOutDir $i --ccs_fofn=reads_of_insert.fofn
```

In distributed jobs, the output is always a single integer so it always works.

In local mode, that's apparently not the case.  

I'm not sure of the proper solution to this bug at the moment, it will require
some deeper research into how P_IsoSeqCluster.py really works.

**Comment by pb-jlandolin** [comment]
The fix will be released in the next 2.3 patch #3 , slated to be out in a few weeks.
